src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5236,37 +5272,34 @@ private BoundExpression BindTypeParameterCreationExpression(ObjectCreationExpres
         {
             AnalyzedArguments analyzedArguments = AnalyzedArguments.GetInstance();
             BindArgumentsAndNames(node.ArgumentList, diagnostics, analyzedArguments);
+            var result = BindTypeParameterCreationExpression(node, typeParameter, analyzedArguments, node.Initializer, node.Type, diagnostics);
+            analyzedArguments.Free();
```

Comment:
>analyzedArguments.Free(); [](start = 12, length = 25)

is the removal of the `try/finally` intentional?

Reply:
removed uniformly in all new helper overloads to reduce noise, especially since we have a single exit point and the method is not too long.